sdk/search/azure-search/src/main/java/com/azure/search/models/IndexBatch.java

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.search.Document;
+import com.azure.search.implementation.models.IndexActionType;
+import com.azure.search.implementation.models.IndexBatchImpl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Contains a batch of document write actions to send to the index.
+ */
+@Fluent
+public class IndexBatch<T> extends IndexBatchImpl<T> {
```

Comment:
It is against general guidelines to have a public class which extends from an implementation class given that it isn't considered public in JDK 9+.

Reply:
IndexBatchImpl moved back under models (removed from implementation) - autorest config will be updated as well as this is a generated class